File: apps/modernization-api/src/main/java/gov/cdc/nbs/patient/investigation/PatientInvestigation.java
package gov.cdc.nbs.patient.investigation;

import com.fasterxml.jackson.annotation.JsonProperty;

public record PatientInvestigation(
    @JsonProperty(required = true) String investigationId,
    @JsonProperty(required = true) long identifier,
    @JsonProperty String startedOn,
    @JsonProperty(required = true) String condition,
    @JsonProperty(required = true) String status,
    @JsonProperty String caseStatus,
    @JsonProperty(required = true) String jurisdiction,
    @JsonProperty String coInfection,
    @JsonProperty String notification,
    @JsonProperty InvestigatorName investigatorName,
    @JsonProperty(required = true) boolean comparable) {

  public record InvestigatorName(
      @JsonProperty String first,
      @JsonProperty String last) {
  }
}

File: apps/modernization-ui/src/apps/patient/file/events/investigations/PatientInvestigation.ts
export type PatientInvestigation = {
    investigationId: string;
    identifier: number;
    startedOn?: Date;
    condition: string;
    status: string;
    caseStatus?: string;
    jurisdiction: string;
    coInfection?: string;
    notification?: string;
    investigatorName?: {
        first?: string;
        last?: string;
    };
    comparable: boolean;
};
